### PR TITLE
 Implemented some series methods for gamma and error functions

### PR DESCRIPTION
**Pull request by Abhishekjsr283204**
References to other Issues or PRs

Brief description of what is fixed or changed

While implementing the leading term for expint(nu, z) function, we don't have to take care of cases where nu < 1, and as the already implemented _eval_nseries method shows we essentially only need to focus on cases where nu >=1.
Because expint(nu, z) when nu < 1 will be rewritten in the forms of exponentials.

Other comments

* series
  * Implemented leading term methods for expint and Si function.
  * Implemented leading term methods for uppergamma and lowergamma function.

### CODE REVIEW

File: sympy/functions/special/error_functions.py [hunk 1]
```diff
@@ -1409,7 +1409,19 @@ def _eval_rewrite_as_Si(self, nu, z, **kwargs):
...
     _eval_rewrite_as_Chi = _eval_rewrite_as_Si
     _eval_rewrite_as_Shi = _eval_rewrite_as_Si
-
+    
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.functions.special.gamma_functions import digamma
+        nu, z = self.args
+        if nu.is_real and nu >= 1:
+            term_one = (-z)**(nu - 1)/factorial(nu - 1) * (digamma(nu) - log(z))
+            term_two = 1/(nu - 1)
```
Comment (VascoSch92) [inline sympy/functions/special/error_functions.py:1418]: you can have division by zero here, right?
You need something like that:
```python
if nu == 1:
     term_two = z/(2 - nu)
else:
    term_two = 1/(nu - 1)
```

Comment (Abhishekjsr283204) [reply sympy/functions/special/error_functions.py:1418]: Thank you for your review. You raised a valid concern about the potential for division by zero in the calculation of term_two. I agree that this could be a problem in certain cases. To address this issue, I'll ensure that nu - 1 is not equal to zero before performing the division. If nu equals 1, I'll handle the calculation differently to avoid division by zero

**Comment by DarkNisa** [comment]
So the main problem here is that you haven't written pull request properly
so it doesn't start testing the whole changes, do you need to rewrite it according 
the guide that sympy bot attached earlier.

File: sympy/functions/special/error_functions.py [hunk 2]
```diff
@@ -1895,6 +1907,19 @@ def _eval_rewrite_as_sinc(self, z, **kwargs):
         t = Symbol('t', Dummy=True)
         return Integral(sinc(t), (t, 0, z))
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        arg = self.args[0].as_leading_term(x)
+        arg0 = arg.subs(x, 0)
+
+        if arg0 is S.NaN:
+            arg0 = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
```
Comment (DarkNisa) [inline sympy/functions/special/error_functions.py:1915]: some thoughts over the code:
`arg0 = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')`
maybe it's better to move dir calculation to temp variable to make it more readable

File: sympy/functions/special/error_functions.py (hunk 1 above)
Comment (DarkNisa) [reply sympy/functions/special/error_functions.py:1418]: some thoughts over the code:
`nu == 1:` the problem here that `term_two = z/(2 - nu)`
is confusing statement and better to be rewritten into `term_two = z`
and also it is confusing having intersected nested `if` statements
`nu >= 1` and `nu == 1` so I think it's better to rewrite it more clearly without such intersections
`arg` variable is unnecessary here and better to be moved to `return` statement
like `return Add(*[term_one, term_two]).as_leading_term(x, logx=logx)`

Comment (anutosh491) [reply sympy/functions/special/error_functions.py:1423]: Correct please add tests for the same (and also add the corner cases being questioned here). It shall help us review much more earily

File: sympy/functions/special/gamma_functions.py [hunk 3]
```diff
@@ -383,12 +383,18 @@ def _eval_aseries(self, n, args0, x, logx):
...
+
     def _eval_rewrite_as_expint(self, s, x, **kwargs):
         from sympy.functions.special.error_functions import expint
         if s.is_integer and s.is_nonpositive:
             return self
         return self.rewrite(uppergamma).rewrite(expint)
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        return self.rewrite(uppergamma)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+
```
Comment (anutosh491) [inline sympy/functions/special/gamma_functions.py:397]: Add tests for this.

File: sympy/functions/special/gamma_functions.py [hunk 4]
```diff
@@ -556,6 +562,10 @@ def _eval_rewrite_as_expint(self, s, x, **kwargs):
         from sympy.functions.special.error_functions import expint
         return expint(1 - s, x)*x**s
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.functions.special.error_functions import expint
+        return self.rewrite(expint)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+
```
Comment (anutosh491) [inline sympy/functions/special/gamma_functions.py:568]: Add tests for this too.

**Comment by anutosh491** [comment]
Also please fix the code quality. I think you might have to pull the latest changes and rebase on top of it.

File: sympy/functions/special/error_functions.py (hunk 1 above)
Comment (Abhishekjsr283204) [reply sympy/functions/special/error_functions.py:1423]: Yes! you are correct i will fix it

File: .mailmap [hunk 5]
```diff
@@ -49,9 +49,9 @@
 #   The mailmap file was reordered
 #
 #   For instructions on updating the .mailmap file see:
-#   https://docs.sympy.org/dev/contributing/new-contributors-guide/workflow-process.html#add-your-name-and-email-address-to-the-mailmap-file
+#   https://docs.sympy.org/dev/contributing/new-contributors-guide/workflow-.html#add-your-name-and-email-address-to-the-mailmap-file
 #
-#   ...
+#   ...process
```
Comment (anutosh491) [inline .mailmap:54]: I don't understand these changes in the mailmap file. I am not sure we need them

File: .mailmap [hunk 6]
```diff
@@ -185,6 +185,8 @@
 #
 # See also: 'git shortlog --help' and 'git check-mailmap --help'.
 #
+
+
```
Comment (anutosh491) [inline .mailmap:189]: Why do we need these lines here ?

File: .mailmap (hunk 5 above)
Comment (Abhishekjsr283204) [reply .mailmap:54]: Oops i will fix it

File: .mailmap (hunk 6 above)
Comment (Abhishekjsr283204) [reply .mailmap:189]: Yes, this is the unnecessary line i will fix it!
